scripts/packages/VSCodeTestServer/src/VSCodeTestServer.jl

```diff
@@ -169,6 +169,33 @@ function run_testitem_handler(conn, params::TestserverRunTestitemRequestParams)
     end
 end
 
+function create_test_message_for_failed(i)
+    (expected, actual) = extract_expected_and_actual(i)
+    return TestMessage(sprint(Base.show, i),
+        expected,
+        actual,
+        Location(filepath2uri(string(i.source.file)), Range(Position(i.source.line - 1, 0), Position(i.source.line - 1, 0))))
+end
+
+function extract_expected_and_actual(result)
+    try
+        if isa(result, Test.Fail)
+            s = result.data
+            if isa(s, String)
+                m = match(r"\"(.*)\" == \"(.*)\"", s)
+                if m !== nothing
+                    expected = unescape_string(m.captures[1])
+                    actual = unescape_string(m.captures[2])
+                    return (expected, actual)
+                end
+            end
+        end
+    catch err
+        # if anything goes wrong (e.g. unescaping failed), just return nothing (meaning no diff will be shown)
+    end
```

Comment:
I'm generally not a fan of try catch blocks that blanket hide errors, I think I would prefer that we just fail if things go wrong, and then hopefully telemetry will tell us about it and we can fix it? Unless this is handling a specific failure mode that you know about, but in that case I think we should also try to handle that more narrowly.

Reply:
I wasn't sure if  result.data exists in older julia versions, but I've now checked, and it exists at least since 1.0.

the only thing that could happen is what is already mentioned in the comment: unescaping might fail if the user registers a weird failure result, e.g.
@testitem "#bad_escape" begin
    Test.record(Test.get_testset(), Test.Fail(:test,"" , "\"\\,\" == \"\"", false, LineNumberNode(2, @__FILE__)))
end

I've pushed an updated commit that narrow the try...catch to those unescape statements.